File: src/main/java/com/hospital/management/appointment/service/AppointmentService.java
package com.hospital.management.appointment.service;

import com.hospital.management.appointment.dto.AppointmentRequestDTO;
import com.hospital.management.appointment.dto.AppointmentResponseDTO;

import java.util.List;

public interface AppointmentService {

    AppointmentResponseDTO saveAppointment(AppointmentRequestDTO dto);

    List<AppointmentResponseDTO> getAllAppointments();

    AppointmentResponseDTO getAppointmentById(Long id);

    AppointmentResponseDTO updateAppointment(Long id, AppointmentRequestDTO dto);

    void deleteAppointment(Long id);
}

File: src/services/appointmentService.js
import api from "./api";

export const getAllAppointments = () => {
  return api.get("/api/appointments");
};

export const saveAppointment = (appointment) => {
  return api.post("/api/appointments", appointment);
};

export const deleteAppointment = (id) => {
  return api.delete(`/api/appointments/${id}`);
  
};
// UPDATE APPOINTMENT
export const updateAppointment = (id, data) => {
  return api.put(`/api/appointments/${id}`, data);
};